test/test_files/function/gds/rec_joins_small.test

```diff
@@ -13,6 +13,27 @@
 
 -STATEMENT CALL enable_gds = true;
 ---- ok
+-STATEMENT MATCH p = (a:person1)-[e:knows11*1..4 (r, n | WHERE n.age < 15)]->(b:person1)
```

Comment:
I didn't look closely at your tests but these tests look cursory. We need more thorough test cases:
1) Covering all queries: so cover single and shortest path queries.
2) For variable length: covering lower bounds that are not 1.
3) Covering when the node predicate applies to some but not all of the edges in an adjacency list. We need to ensure there are cases when there are multiple parents p1, .., pk, of a node u and that sometimes all of them pass pass the filter, sometimes none, and sometimes some.
4) Further we should test cyclic cases when the source does not pass the filter but we still output the paths when the source node is actually the source and not an intermediate.

Let's talk about your test cases. We need to be very rigorous here.